src/microbe_stage/Microbe.Contact.cs

```diff
@@ -383,16 +424,124 @@ public bool CanEngulf(Microbe target)
             return false;
         }
 
+        // Limit amount of things that can be engulfed at once
+        if (IngestedSizeCount > EngulfSize || IngestedSizeCount + target.EngulfSize > EngulfSize)
+            return false;
+
+        // Too many things attempted to be engulfed at once
+        if (attemptingToEngulf.Sum(e => e.EngulfSize) + target.EngulfSize > EngulfSize)
+            return false;
+
         // Disallow cannibalism
-        if (target.Species == Species)
+        if (isMicrobe && microbe!.Species == Species)
             return false;
 
         // Membranes with Cell Wall cannot engulf
         if (Membrane.Type.CellWall)
             return false;
 
         // Needs to be big enough to engulf
-        return EngulfSize >= target.EngulfSize * Constants.ENGULF_SIZE_RATIO_REQ;
+        return EngulfSize > target.EngulfSize * Constants.ENGULF_SIZE_RATIO_REQ;
+    }
+
+    public void OnAttemptedToBeEngulfed()
+    {
+        Membrane.WigglyNess = 0;
+
+        // Make the render priority of our organelles be on top of the highest possible render priority
+        // of the hostile engulfer's organelles
+        var hostile = HostileEngulfer.Value;
+        if (hostile != null)
+        {
+            foreach (var organelle in organelles!)
+            {
+                var newPriority = Mathf.Clamp(Hex.GetRenderPriority(organelle.Position) +
+                    hostile.organelles!.MaxRenderPriority, 0, Material.RenderPriorityMax);
+                organelle.UpdateRenderPriority(newPriority);
+            }
+        }
+
+        // Just in case
+        playerEngulfedDeathTimer = 0;
+    }
+
+    public void OnIngestedFromEngulfment()
+    {
+        OnIngestedByHostile?.Invoke(this, HostileEngulfer.Value!);
+    }
+
+    public void OnExpelledFromEngulfment()
+    {
+        var hostile = HostileEngulfer.Value;
+
+        // Reset wigglyness
+        ApplyMembraneWigglyness();
+
+        // Reset our organelles' render priority back to their original values
+        foreach (var organelle in organelles!)
+        {
+            organelle.UpdateRenderPriority(Hex.GetRenderPriority(organelle.Position));
+        }
+
+        // Our engulfer also has its own engulfer and it wants to claim us
+        if (hostile?.HostileEngulfer.Value != null)
```

Comment:
Shouldnt' the hostile engulfer's engulfer handle this? To me this kind of looks like the wrong object here is trying to do the work, which complicates this method.

Reply:
Fixed.